src/common/src/array/stream_chunk.rs

```diff
@@ -174,6 +174,40 @@ impl StreamChunk {
         StreamChunk::new(new_ops, columns)
     }
 
+    /// Split the `StreamChunk` into multiple chunks with the given size at most.
+    ///
+    /// When the total cardinality of all the chunks is not evenly divided by the `size`,
+    /// the last new chunk will be the remainder.
+    ///
+    /// For `UpdateDelete` and `UpdateInsert`, they will be kept in once chunk. As a result,
+    /// some chunks may have `size + 1` rows.
+    pub fn split(&self, size: usize) -> Vec<Self> {
+        let data_types = self.data_types();
+        let mut rows = vec![];
+        let mut results = vec![];
+
+        let mut iter = self.rows();
+        while let Some(row) = iter.next() {
+            rows.push(row);
+            if rows.len() == size {
+                if rows.last().unwrap().0 == Op::UpdateDelete {
+                    let next_row = iter.next().unwrap();
```

Comment:
```suggestion
                    let next_row = iter.next().expect("UpdateDelete should have UpdateInsert");
```

Reply:
Unrelated question: Why do we have `UpdateDelete` and `UpdateInsert` instead of keeping them as a unit operation `Update`. Seems a lot safer to do so.

Reply:
> Unrelated question: Why do we have `UpdateDelete` and `UpdateInsert` instead of keeping them as a unit operation `Update`. Seems a lot safer to do so.

Yeah, I am also considering adding a new function to return `(UpdateDelete, UpdateInsert)` as whole.

Reply:
I believe there are other issues where `UpdateDelete` and `UpdateInsert` caused some breakage, but I can't recall the issue.

I wonder why original design choose to separate `Update` into 2 parts.

Reply:
> I wonder why original design choose to separate Update into 2 parts.

Because an `Update` needs to carry both values before and after the update. As we are using a column-oriented memory stucture, do we have any other choices?

```
Example: to represent Update((1,'foo')->(2,'bar'))
+----+---+-----+
| U- | 1 | foo |
| U+ | 3 | bar | 
+----+---+-----+
(Obviously, it takes 2 rows to store such `Update`)
```